skillmap/src/App.tsx

```diff
@@ -0,0 +1,294 @@
+                    <div className="skill-map-container">
+                        { error
+                            ? <div className="skill-map-error">{error}</div>
+                            : maps?.map((el, i) => {
+                                return <SkillCarousel map={el} key={i} />
+                            })}
+                    </div>
+                </div>
+                }
+                <AppModal />
+            </div>);
+    }
+
+    protected applyQueryFlags(user: UserState, maps?: SkillMap[], sourceUrl?: string) {
+        if (this.queryFlags["debugNewUser"] === "true") {
+            user.isDebug = true;
+            user.mapProgress = {};
+            user.completedTags = {};
+        }
+
+        if (this.queryFlags["debugCompleted"] === "true") {
+            user.isDebug = true;
+
+            if (maps) {
+                for (const map of maps) {
+                    user.mapProgress[map.mapId] = {
+                        mapId: map.mapId,
+                        activityState: {}
+                    };
+
+                    for (const key of Object.keys(map.activities)) {
+                        const activity = map.activities[key];
+                        if (!user.mapProgress[map.mapId].activityState[activity.activityId]) {
+                            user.mapProgress[map.mapId].activityState[activity.activityId] = {
+                                activityId: activity.activityId,
+                                isCompleted: true
+                            };
+                        }
+                        else {
+                            user.mapProgress[map.mapId].activityState[activity.activityId].isCompleted = true;
+                        }
+
+                        if (activity.tags?.length && sourceUrl) {
+                            for (const tag of activity.tags) {
+                                if (!user.completedTags[sourceUrl][tag]) user.completedTags[sourceUrl][tag] = 0;
+                                user.completedTags[sourceUrl][tag]++;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    protected onStoreChange = async () => {
+        const { user } = store.getState();
+
+        if (user !== this.loadedUser && (!this.loadedUser || user.id === this.loadedUser.id)) {
+            await saveUserStateAsync(user);
+            this.loadedUser = user;
+        }
+    }
+}
+
+function mapStateToProps(state: SkillMapState, ownProps: any) {
+    if (!state) return {};
+    return {
+        skillMaps: state.maps,
+        activityOpen: !!state.editorView
+    };
+}
+
+async function updateLocalizationAsync(targetId: string, baseUrl: string, code: string, pxtBranch: string, targetBranch: string, live?: boolean, force?: boolean) {
```

Comment:
nit: I'm assuming we're duping this portion to avoid refactoring it into in pxtlib/util's version? Maybe worth adding a note here and in the original definition pointing to eachother / that if we change anything in the structure of localizations (beginning / end of fn) we should propagate it to both sections